### ISSUE TITLE
Move twisted.mail.smtp interfaces into a common interfaces file

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8786]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8786|
|Type|enhancement|
|Created|2016-08-25 15:05:20Z|
|Branch|https://github.com/twisted/twisted/tree/8786-smtp-interfaces|

See [#8785](https://github.com/twisted/twisted/issues/8785)

<details><summary>Searchable metadata</summary>

```
trac-id__8786 8786
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8786_smtp_interfaces 8786-smtp-interfaces
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1472137520822512 1472137520822512
changetime__1473355793665239 1473355793665239
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Happy with this; builders are green and coverage is at 100%.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

PR: https://github.com/twisted/twisted/pull/525

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @hawkowl|
|-|-|

https://github.com/twisted/twisted/pull/525#issuecomment-242631993

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 64b820afd8d5d935cbe7146dd294108d4da75f94
```
#!CommitTicketReference repository="" revision="64b820afd8d5d935cbe7146dd294108d4da75f94"
Merge 8786-smtp-interfaces: Move twisted.mail.smtp interfaces into a common file

Author: hawkowl
Reviewer: rodrigc
Fixes: #8786
```
